### ISSUE TITLE
Question about adding events

### ISSUE THREAD

**Issue by nakeddogg** [neowutran/ShinraMeter#494]
Hello buddies,

I tried to reach someone who knows more than me about the events script, but I could not find any message function in Github. So, I'm asking my question through this community.  If I make you upset, I apologize.

I just heard from one of my friends that if a user sets the event script, Shinrameter lets users know when the monster they looking for spawn on the location where the users are.

In other words, they don't even need to explore or roam the map to find the monster.

If there is any trigger script about this or any sample, may I ask you guys to teach me?

Thank you.
Best regards,

**Comment by Risenio** [comment]
Hey there.

What you're talking about is a Tera-Toolbox or Tera-Proxy module which does notify you (could be in-game or using Windows Toast notifications) about spawns of certain NPCs spawns.
• Notification method is done when the NPC spawns in your visual client range (could be increased via Options.Video.['Character quality'] => 'PC view distance'), your visual range is not only looking direction but all around you, so what module does just save you looking around (all directions up down left right front back, etc.), it doesn't roam the map for you unless you're talking about Botting actions.
• Other notification could be linked to system messages which are server-wise announced regarding spawn, but these doesn't specify location, only region/province.

○ Shinra by default doesn't have such function, unless you re-write couple parts of its core source.

○ Shinra's events and what they can do is [written here](https://github.com/neowutran/ShinraMeter/wiki/Events) in details, feel free to have a look, more code examples can be [found here](https://github.com/neowutran/ShinraMeter/tree/master/Lang/Resources/en).
